### PR TITLE
Fix manager governance instantiation/updates

### PR DESCRIPTION
**Pull request by CyberHoward**
Improves the `GovernanceDetails` type to be used consistently to describe the governance structure of the OS. 

Also makes the type non-exhaustive so we can add new variants in the future if we want!

### CODE REVIEW

File: packages/abstract-core/src/objects/gov_type.rs [hunk 1]
```diff
@@ -1,54 +1,141 @@
...
+
+const MIN_GOV_TYPE_LENGTH: usize = 4;
+const MAX_GOV_TYPE_LENGTH: usize = 64;
+
 /// Governance types
 #[derive(Deserialize, Serialize, Clone, Debug, PartialEq, Eq, JsonSchema)]
-
-pub enum GovernanceDetails {
+#[non_exhaustive]
+pub enum GovernanceDetails<T: AddressLike> {
```
Comment (adairrr) [inline packages/abstract-core/src/objects/gov_type.rs:16]: Nice addition of the `AddressLike`!

File: packages/abstract-core/src/objects/gov_type.rs [hunk 2]
```diff
@@ -1,54 +1,141 @@
...
+use crate::AbstractError;
+
+const MIN_GOV_TYPE_LENGTH: usize = 4;
+const MAX_GOV_TYPE_LENGTH: usize = 64;
+
 /// Governance types
 #[derive(Deserialize, Serialize, Clone, Debug, PartialEq, Eq, JsonSchema)]
-
-pub enum GovernanceDetails {
+#[non_exhaustive]
```
Comment (adairrr) [inline packages/abstract-core/src/objects/gov_type.rs:15]: Good call on this. Would an example for additional "hard-coded" governance type be DaoDao or Vectis for example?

File: packages/abstract-core/src/objects/gov_type.rs [hunk 3]
```diff
@@ -1,54 +1,141 @@
...
+                    || governance_type.len() > MAX_GOV_TYPE_LENGTH
+                {
+                    return Err(AbstractError::FormattingError {
+                        object: "governance_type".to_string(),
+                        expected: format!(
+                            "a string of length between {} and {}",
+                            MIN_GOV_TYPE_LENGTH, MAX_GOV_TYPE_LENGTH
+                        ),
+                        actual: format!("a string of length {}", governance_type.len()),
+                    });
```
Comment (adairrr) [inline packages/abstract-core/src/objects/gov_type.rs:54]: As we are doing similar validations for length we could pull this out into a `ValidatedString` or other. 

And additionally - should we disallow spaces / other chars?

File: packages/abstract-core/src/objects/gov_type.rs (hunk 2 above)
Comment (CyberHoward) [reply packages/abstract-core/src/objects/gov_type.rs:15]: could be, or `SelfOwned` in the case where we set the proxy as the owner! 
Account inception...

File: packages/abstract-core/src/objects/gov_type.rs (hunk 3 above)
Comment (CyberHoward) [reply packages/abstract-core/src/objects/gov_type.rs:54]: in a name, hmm maybe allow it? Pulling it into a type is kinda overkill imo.

Comment (adairrr) [reply packages/abstract-core/src/objects/gov_type.rs:54]: true true. Point still stands about the spaces.

Comment (CyberHoward) [reply packages/abstract-core/src/objects/gov_type.rs:54]: we can do all kinds of verification like mentioned in the report. The question is: do we really want to include that logic in the on-chain contract? These checks are easier and cheaper to perform in the front-end and the strings (be it name, governance type, etc. don't effect anything on-chain. 

The line I would draw is that the checks should be performed on contracts that are low-use and are rarely loaded, that way the contract's binary size doesn't impact gas costs of module calls.

Comment (CyberHoward) [reply packages/abstract-core/src/objects/gov_type.rs:54]: format gov-type as module namespaces

File: contracts/account/manager/src/commands.rs [hunk 4]
```diff
@@ -232,30 +233,29 @@ pub fn uninstall_module(deps: DepsMut, msg_info: MessageInfo, module_id: String)
...
+    OWNER.execute_update_admin::<Empty, Empty>(deps, info, Some(new_owner_addr.clone()))?;
     Ok(ManagerResponse::new(
         "update_owner",
         vec![
             ("previous_owner", previous_owner.to_string()),
-            ("owner", new_owner),
+            ("new_owner", new_owner_addr.to_string()),
         ],
     ))
 }
```
Comment (adairrr) [inline contracts/account/manager/src/commands.rs:261]: We should also include the governance as an attribute in the resp.

File: contracts/native/account-factory/src/commands.rs [hunk 5]
```diff
@@ -64,23 +66,14 @@ pub fn receive_cw20(
 pub fn execute_create_account(
     deps: DepsMut,
     env: Env,
-    governance: GovernanceDetails,
+    governance: GovernanceDetails<Addr>,
     _asset: Option<Asset>,
```
Comment (adairrr) [inline contracts/native/account-factory/src/commands.rs:70]: Is this asset supposed to be a default base_asset?

Comment (CyberHoward) [reply contracts/native/account-factory/src/commands.rs:70]: This functionality will be removed (think you might have already set that task to complete, but it isn't)

Comment (CyberHoward) [reply contracts/native/account-factory/src/commands.rs:70]: https://app.clickup.com/t/861mkce4z this one!

Comment (CyberHoward) [reply contracts/native/account-factory/src/commands.rs:70]: re-opened the issue

**Review by adairrr — APPROVED** [review]

**Comment by adairrr** [comment]
Task linked: [CU-861mkce9g Preserve Governance type enum on manager init and update](https://app.clickup.com/t/861mkce9g)

**Pull request merged**